2-planning/0-introduction.adoc

```diff
@@ -0,0 +1,56 @@
+= Introduction
+
+When I was younger and dumber, the planning stage for APIs was pretty much
+"think of a list of endpoints then start coding". This is wildly insufficient.
+As the <<theory-contracts,API Contracts>> chapter explained, contracts are
+incredibly important throughout the life-cycle of an API. We are going to look
+into how they can be used in the planning and design stage.
+
+Most of the planning and design phase is intended to do two things:
+
+1. Make developers think really hard about the requirements for their API 2.
+Gather feedback from clients application developers 3. Use the feedback to
+improve the planning before the development phase begins
+
+== Time is Money
+
+Poorly planned APIs waste a lot of time. Developers cost money, so paying extra
+money for developers to rewriting a bunch of code they wrote just recently due
+to some oversight is a cost most companies would like to avoid. I've seen APIs
+go from v1 to v2 in the first few months, because the developers handed over a
+finished product which might have been ok for one client, but caused some issues
+for another client.
+
+If that client had been involved earlier on then a solution could have been
+found to satisfy the needs of both clients, and the v1 could have been used for
+both. Now we've got two separate versions of the API being used at the same
+time, with developers maintaining two code paths, and at some point the original
+client using the original version of the API is going to have to rewrite their
+code to use the newer version of the API. What a daft mess.
+
+== Gather Feedback Early
+
+If clients had more insight into the planning of the API, they have a much
+better chance of catching out issues earlier on. We can give them fake API
```

Comment:
Who is "We"? Who writes the API contract, who creates the mock? Is the developer responsible to write the contract? Should a company have API specialists?

Reply:
@chchrist hey, I've added a commit which answers your question. Throughout the book im saying "we" as conversational, literally you and me.

Reply:
ah, cool 👍